desktop/core/ext-py3/boto-2.49.0/boto/gs/bucket.py

```diff
@@ -273,9 +273,9 @@ def delete_key(self, key_name, headers=None, version_id=None,
         query_args_l = []
         if generation:
             query_args_l.append('generation=%s' % generation)
-        self._delete_key_internal(key_name, headers=headers,
-                                  version_id=version_id, mfa_token=mfa_token,
-                                  query_args_l=query_args_l)
+        return self._delete_key_internal(key_name, headers=headers,
+                                         version_id=version_id, mfa_token=mfa_token,
+                                         query_args_l=query_args_l)
```

Comment:
So that it returns the Key object which got deleted. IIRC we then use the returned Key object to check if it exist in the FS or not to verify if the delete operation was successful or not.

It is returning it for S3 bucket implementation but somehow this change was missed for GS in the boto package code.

Reply:
Unless I'm misunderstanding something here I think that we should try to avoid these kind of modifications since they are very hard to track. Think about what happens when another developer upgrades the boto lib in the future, this change will be lost and break functionality. 

Is it not possible to check if a file has been deleted without this return value?
If this change is really needed can you solve it by creating a custom Hue wrapper around the boto library that has its own delete function or by creating a custom version of this file that signals that it has been modified (including very clear comments on what was changed).

Reply:
@bjornalm - Agreed, I thought about it. Boto2 is already deprecated from long time. v2.49.0 is the last and latest release for it.

Yes your understanding it correct that if someone tries to modify such files in ext-py3 directory then they can miss these changes but this issues is with every other package in ext-py3 directory.

The ideal long term solution is to upgrade to boto3 which is a big task and unfortunately that does not support GS. So seeing the timelines, we went with boto2 for GS also here.

In the long term, switching to boto3 for S3 and some other maintained package like `google-cloud-storage` maybe for GS (if it suits our need) makes sense for sure.

The above change should not have any side effect because Hue should be the only one using the methods of the GS Bucket class and GS Key class.

Reply:
I would normally prefer changes to libs be a separate commit if there is no way out. This can help us reapply the same diffs in cases of upgrades.

Reply:
Makes sense, I have pushed the changes as a separate PR: https://github.com/cloudera/hue/pull/3484

Please check it out.